### PR TITLE
servermaster(engine): add backoff config, fix timeout in chaos test

### LINKED ISSUE #7005: node failure and chaos test could suffer timeout after enabling backoff mechanism
### What did you do?

ref: https://ci2.pingcap.net/blue/organizations/jenkins/engine_ghpr_integration_test/detail/engine_ghpr_integration_test/379/pipeline

detail logs: [log-e2e_node_failure-dm_many_tables.tar.gz](https://github.com/pingcap/tiflow/files/9523181/log-e2e_node_failure-dm_many_tables.tar.gz)

### What did you expect to see?

_No response_

### What did you see instead?

- Case failed at 2022/09/07 14:27:09.150 +00:00

```go
[2022/09/07 14:27:09.150 +00:00] [WARN] [e2e_node_chaos_test.go:58] ["check fake job failed"] [error="value not equals, expected: 'restart-executor-value-2', actual: 'restart-executor-value-1', checkpoint {\"ticks\":{\"0\":596,\"1\":1006,\"2\":595,\"3\":1007},\"checkpoints\":{\"0\":{\"tick\":0,\"revision\":50,\"mvcc-count\":6,\"value\":\"restart-executor-value-1\"},\"1\":{\"tick\":0,\"revision\":51,\"mvcc-count\":6,\"value\":\"restart-executor-value-1\"},\"2\":{\"tick\":0,\"revision\":52,\"mvcc-count\":6,\"value\":\"restart-executor-value-1\"},\"3\":{\"tick\":0,\"revision\":53,\"mvcc-count\":6,\"value\":\"restart-executor-value-1\"}}}"] [errorVerbose="value not equals, expected: 'restart-executor-value-2', actual: 'restart-executor-value-1', checkpoint {\"ticks\":{\"0\":596,\"1\":1006,\"2\":595,\"3\":1007},\"checkpoints\":{\"0\":{\"tick\":0,\"revision\":50,\"mvcc-count\":6,\"value\":\"restart-executor-value-1\"},\"1\":{\"tick\":0,\"revision\":51,\"mvcc-count\":6,\"value\":\"restart-executor-value-1\"},\"2\":{\"tick\":0,\"revision\":52,\"mvcc-count\":6,\"value\":\"restart-executor-value-1\"},\"3\":{\"tick\":0,\"revision\":53,\"mvcc-count\":6,\"value\":\"restart-executor-value-1\"}}}\ngithub.com/pingcap/tiflow/engine/test/e2e.(*ChaosCli).CheckFakeJobKey\n\t/home/jenkins/agent/workspace/engine_ghpr_integration_test/go/src/github.com/pingcap/tiflow/engine/test/e2e/e2e_test_cli.go:199\ngithub.com/pingcap/tiflow/engine/test/e2e_test.updateKeyAndCheckOnce.func1\n\t/home/jenkins/agent/workspace/engine_ghpr_integration_test/go/src/github.com/pingcap/tiflow/engine/test/e2e/e2e_node_chaos_test.go:56\ngithub.com/stretchr/testify/assert.Eventually.func1\n\t/go/pkg/mod/github.com/stretchr/testify@v1.8.0/assert/assertions.go:1749\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1571"]
    e2e_node_chaos_test.go:51: 
        	Error Trace:	/home/jenkins/agent/workspace/engine_ghpr_integration_test/go/src/github.com/pingcap/tiflow/engine/test/e2e/e2e_node_chaos_test.go:51
        	            				/home/jenkins/agent/workspace/engine_ghpr_integration_test/go/src/github.com/pingcap/tiflow/engine/test/e2e/e2e_node_chaos_test.go:166
        	Error:      	Condition never satisfied
        	Test:       	TestNodeFailure
--- FAIL: TestNodeFailure (190.36s)
```

- Restart executor at 2022/09/07 14:26:11.145 +00:00

```go
[2022/09/07 14:26:10.349 +00:00] [INFO] [e2e_test_cli.go:222] ["Start executing command"] [cmd="/usr/bin/docker restart server-executor-2"]
[2022/09/07 14:26:11.145 +00:00] [INFO] [e2e_test_cli.go:235] ["Finished executing command"] [cmd="/usr/bin/docker restart server-executor-2"] [output="server-executor-2\n"]
[2022/09/07 14:26:11.145 +00:00] [INFO] [e2e_test_cli.go:243] ["Finished restarting container"] [name=server-executor-2]
[2022/09/07 14:26:11.145 +00:00] [DEBUG] [e2e_node_chaos_test.go:42] ["update fake job key"] [job-id=8acbd7fc-27df-410b-8712-454904726f14] [update-value=restart-executor-value-2] [expect-mvcc=7]
```

- Update fake job key and check value is updated in 60s, finally timeout.

- Job master offline and backoff works

```go
[2022/09/07 14:26:30.797 +00:00] [INFO] [jobmanager.go:689] ["on worker offline"] [id=8acbd7fc-27df-410b-8712-454904726f14] [reason="[DFLOW:ErrWorkerOffline]worker is offline: workerID: 8acbd7fc-27df-410b-8712-454904726f14, error message:
[2022/09/07 14:26:30.799 +00:00] [DEBUG] [callbacks.go:134] ["trace log"] [component=gorm] [elapsed=2.02234ms] [sql="DELETE FROM `worker_statuses` WHERE job_id = 'dataflow-engine-job-manager' AND id = '8acbd7fc-27df-410b-8712-454904726f14
[2022/09/07 14:26:30.799 +00:00] [INFO] [worker_handle.go:150] ["Worker tombstone is cleaned"] [worker-id=8acbd7fc-27df-410b-8712-454904726f14]
[2022/09/07 14:26:30.799 +00:00] [INFO] [backoff.go:189] ["job backoff interval is changed"] [job-id=8acbd7fc-27df-410b-8712-454904726f14] [old-interval=0s] [new-interval=14.405692858s]
```

- Job master failover happens at 2022/09/07 14:26:45.796 +00:00, which costs `34s` after executor is restarted.

```go
[2022/09/07 14:26:45.796 +00:00] [INFO] [master.go:646] [CreateWorker] [job_id=dataflow-engine-job-manager] [worker-type=3] [worker-config="{\"seq-id\":2,\"created-at\":\"2022-09-07T14:24:00.805Z\",\"updated-at\":\"2022-09-07T14:24:38.114Z\",\"project-id\":\"dfe_proj_default\",\"id\":\"8acbd7fc-27df-410b-8712-454904726f14\",\"type\":3,\"state\":2,\"node-id\":\"executor-server-executor-0:10241-0c408b9d\",\"addr\":\"server-executor-0:10241\",\"epoch\":12,\"config\":\"eyJqb2ItbmFtZSI6InRlc3Qtbm9kZS1mYWlsdXJlIiwid29ya2VyLWNvdW50Ijo0LCJ0YXJnZXQtdGljayI6MTAwMDAwMDAsImV0Y2Qtd2F0Y2gtZW5hYmxlIjp0cnVlLCJldGNkLWVuZHBvaW50cyI6WyJldGNkLXN0YW5kYWxvbmU6MjM3OSJdLCJldGNkLXdhdGNoLXByZWZpeCI6Ii9mYWtlLWpvYi90ZXN0LyIsImV0Y2Qtc3RhcnQtcmV2aXNpb24iOjAsImluamVjdC1lcnJvci1pbnRlcnZhbCI6MH0=\",\"error-message\":\"\",\"detail\":null,\"ext\":{\"selectors\":null},\"Deleted\":null}"] [cost=1] [resources="[]"] [master-id=dataflow-engine-job-manager]
```

- Job recovers at 2022/09/07 14:27:09.578 +00:00, 

```go
[2022/09/07 14:27:09.578 +00:00] [INFO] [fake_master.go:347] ["FakeMaster: Tick"] [status="{\"2edd6f3f-7b3f-4607-9f5a-bc542ae151b2\":{\"business-id\":1,\"tick\":1066,\"checkpoint\":{\"tick\":0,\"revision\":55,\"mvcc-count\":7,\"value\":\"restart-executor-value-2\"}},\"83ff5321-399d-43a0-8e23-921c1a899b7c\":{\"business-id\":3,\"tick\":1068,\"checkpoint\":{\"tick\":0,\"revision\":57,\"mvcc-count\":7,\"value\":\"restart-executor-value-2\"}},\"c3d845b8-5509-4889-9689-dbaec2139aab\":{\"business-id\":2,\"tick\":656,\"checkpoint\":{\"tick\":0,\"revision\":56,\"mvcc-count\":7,\"value\":\"restart-executor-value-2\"}},\"d8de0e7e-79d1-4812-ba7f-1cbf0b399fda\":{\"business-id\":0,\"tick\":658,\"checkpoint\":{\"tick\":0,\"revision\":54,\"mvcc-count\":7,\"value\":\"restart-executor-value-2\"}}}"]
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

master@https://github.com/pingcap/tiflow/commit/690c620898e6aad00e5c6cd49f4c9d332d2e9840

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #7005

### What is changed and how it works?

- Add job backoff config at server level
- Use small interval of backoff configs in integration test

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by amyangfei** [comment]
chaos test passed. https://github.com/pingcap/tiflow/actions/runs/3014351764

**Review by maxshuang — APPROVED** [review]

File: deployments/engine/docker-compose/config/master.toml [hunk 1]
```diff
@@ -12,5 +12,10 @@ schema = "test_business"
 user = "root"
 password = ""
 
-[log] 
+[log]
 level = "debug"
+
+[job-backoff]
```
Comment (CharlesCheung96) [inline deployments/engine/docker-compose/config/master.toml:18]: Whether the configuration is required only during testing?

File: engine/servermaster/jobop/backoff.go [hunk 2]
```diff
@@ -92,11 +82,12 @@ type backoffEvent struct {
 
 // NewJobBackoff creates a new job backoff
 func NewJobBackoff(jobID string, options ...BackoffOption) *JobBackoff {
+	defaultConfig := NewDefaultBackoffConfig()
```
Comment (gozssky) [inline engine/servermaster/jobop/backoff.go:85]: It seems `config` and `BackoffOption` have the same effect. Can we just remove the `BackoffOption`?

File: deployments/engine/docker-compose/config/master.toml (hunk 1 above)
Comment (amyangfei) [reply deployments/engine/docker-compose/config/master.toml:18]: Yes, currently we have default backoff configuration. Later we will align requirements with PM for these values.

File: engine/servermaster/jobop/backoff.go (hunk 2 above)
Comment (amyangfei) [reply engine/servermaster/jobop/backoff.go]: Removed `BackoffOption`, PTAL.

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**